keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/AutoDiffBroadcast.java

```diff
@@ -0,0 +1,54 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.diff;
+
+import io.improbable.keanu.tensor.TensorShape;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+
+import java.util.Arrays;
+
+/**
+ * This class is meant to help with auto diff in operations that support implicit broadcasting. E.g. In
+ * addition/subtraction/multiplication/division scalar operands can be operated with non-scalar operands.
+ */
+public class AutoDiffBroadcast {
+
+    public static PartialDerivative correctForScalarPartialForward(PartialDerivative partial, long[] partialOfShape, long[] targetOfShape) {
+
+        if (shouldCorrectPartialForScalarForward(partial, partialOfShape, targetOfShape)) {
+
+            long[] wrtShape = partial.getWrtShape(partialOfShape);
+            DoubleTensor correctedPartial = DoubleTensor
+                .zeros(TensorShape.concat(targetOfShape, wrtShape))
+                .plus(partial.getPartial());
```

Comment:
not a big fan of `getPartial()` - `getDerivative()`?

Reply:
How about just get() since it's already a PartialDerivative?

Reply:
aye